tests/run/toplevel-implicits/defs_1.scala

```diff
@@ -0,0 +1,15 @@
+package implicits
+
+case class C()
+
+implicit object Cops {
+  def (x: C) pair (y: C) = (x, y)
+}
+
+class D {
+  override def toString = "a D"
+}
+
+implicit class Ddeco(x: D) {
+  def pair(y: D) = (x, y)
+}
```

Comment:
I believe we should also test top-level `implicit def`s (both conversions and non-conversions) and `implicit val`s.